internal/server/answers.go

```diff
@@ -95,6 +92,12 @@ func ginAnswersHandler(s *Server) func(c *gin.Context) {
 				return errors.New("trying to save data on a non-hit step")
 			}
 
+			timeExtension := stepRun.EndedAt.Add(time.Minute * time.Duration(step.HitArgs.Timeout))
```

Comment:
I think we should leave this a fixed time of 1 hour.

Also, what happens if the stepRun hasn't ended yet. EndedAt won't be defined, will it? For sure you can't submit before StartedAt exists, so maybe check that first ; then check if EndedAt exists, if not it's still ongoing, it's good ; then finally accept if < endedAt + 1h.